check_version_status/index.js

```
@@ -0,0 +1,79 @@
+"use strict"
+
+// Inputs or Arguments required
+// JIRA Username
+// JIRA API token
+// JIRA Project ID
+// JIRA Ratehub atlassian Domain
+// Release Identifier - current or next release
+
+const core = require('@actions/core');
+const JiraApi = require('jira-client');
+const semver = require('semver');
+
+const jiraUser = core.getInput('jira-username');
+const jiraApiToken = core.getInput('jira-api-token');
+const latestTag = core.getInput('latest-git-tag');
+const atlassianDomain = core.getInput('atlassian-domain');
+const projectId = core.getInput('jira-project-id');
+const release = core.getInput('which-release');
+
+const jira = new JiraApi({
+  protocol: 'https',
+  host: `${atlassianDomain}`,
+  username: `${jiraUser}`,
+  password: `${jiraApiToken}`,
+  apiVersion: '3',
+  strictSSL: true
+});
+
+function getSemVersion(currentVersion) {
+    const trivialPatchBump = semver.clean(currentVersion);
+    const minorVer = `${semver.major(trivialPatchBump)}.${semver.minor(trivialPatchBump)}.${semver.patch(trivialPatchBump)}`;
+    const releaseBranchName = `release-v${minorVer}`;
+    return { trivialPatchBump , minorVer, releaseBranchName };
+}
+
+function incrementMinorVersion(currentVersion, tagIdentifier) {
+    const bumpedVersion = semver.inc(currentVersion, tagIdentifier);
+    const updatedVer = `${semver.major(bumpedVersion)}.${semver.minor(bumpedVersion)}.${semver.patch(bumpedVersion)}`;               
+    return { bumpedVersion, updatedVer };
+}
+
+function finalReleaseVersion(){
+    if (release == 'current'){
```

So, I'm unfamiliar with the framework for Actions, but it looks like there is no validation on the input.  It is marked as `required` here check_version_status/action.yml but nothing else.  Since JS is a dynamic language you need to add your own constraints.  You handle 2 cases `current` and `next` but don't handle the case for other values including empty strings, undefined, typos, or completely different values.  Outside the 2 handled options what happens with invalid input?
Below you check if the returned value matches a project version and then whether this "find" operation result is defined but I don't think it provides any feedback on the source of the problem (the inputs).  The project lookup may fail (again that too may be due to invalid input or perhaps a network or server error) and so the errors are all compounded with no context.
You should at least log the error condition here (and where ever possible)  so it will be easier to track down the cause(s).  It is always better to fail early (immediately) than to pass an error on (in my opinion this is one of the big faults with the way exception handling is often used.)
This no doubt applies to all 3 of the Actions.

---

will update the actions to handle other cases and exceptions.

---

Will be resolved in another PR.